pilot/pkg/dns/dns.go

```diff
@@ -215,16 +219,18 @@ func (h *LocalDNSServer) ServeDNS(proxy *dnsProxy, w dns.ResponseWriter, req *dn
 		_ = w.WriteMsg(response)
 		return
 	}
+
 	// We did not find the host in our internal cache. Query upstream and return the response as is.
-	response = h.queryUpstream(proxy.upstreamClient, req, log)
-	// Compress the response - we don't know if the incoming response was compressed or not. If it was,
-	// but we don't compress on the outbound, we will run into issues. For example, if the compressed
-	// size is 450 bytes but uncompressed 1000 bytes now we are outside of the non-eDNS UDP size limits
-	response.Truncate(size(proxy.protocol, req))
 	log.Debugf("response for hostname %q (found=false): %v", hostname, response)
+	response = h.queryUpstream(proxy.upstreamClient, req, log)
```

Comment:
why did we change the truncation?

Reply:
Let me check. I did not change intentionally

Reply:
It is removed by mistake. Reverted it